### PR TITLE
Bug fix: `Loader.__len__` returns wrong length

### PR DESCRIPTION
**Pull request by felix0097**
In the current implementation `Loader.__len__` returns `n_obs` instead to the number of batches. This PR should fix this

### CODE REVIEW

**Review by ilan-gold — COMMENTED** [review]
Can we add a test for this now? I would think one of the old `torch` tests were broken silently?

File: tests/test_dataset.py [hunk 1]
```diff
@@ -323,6 +323,45 @@ def test_drop_last(adata_with_zarr_path_same_var_space: tuple[ad.AnnData, Path],
...
+        chunk_size=chunk_size,
+        preload_nchunks=preload_nchunks,
+        batch_size=batch_size,
+        preload_to_gpu=False,
+        to_torch=False,
+        drop_last=drop_last,
+    )
+    loader.add_dataset(**data)
+
+    expected_len = n_obs // batch_size if drop_last else math.ceil(n_obs / batch_size)
```
Comment (ilan-gold) [inline tests/test_dataset.py]: Put this as a parametrized number explicitly.  Also not sure why we're parametrizing on chunk size or number of chunks to preload.  Can you remove? Seems unrelated

Comment (felix0097) [reply tests/test_dataset.py]: It's supposed to also check the `n_iters` function of the `ChunkSampler`. This covers different edge cases. I can remove it, if we only wanna check the `n_iters`.

**Review by ilan-gold — APPROVED** [review]
